docs/model_access_control.md

```diff
@@ -0,0 +1,652 @@
+- `private`: Only the model owner or an admin user can access this model group.
+- `restricted`: The owner, an admin user, or any user who shares one of the model group's backend roles can access any model in this model group. When creating a `restricted` model group, the owner must attach one or more of the owner's backend roles to the model. For a user to be able to access a model group, one of the user's backend roles must match one of the model group's backend roles. This grants the user permissions associated with the user role to which that backend role is mapped. For example, if the backend role is mapped to the `ml_full_access` user role, users with this role can use the Register, Deploy, Predict, Delete, Search and Get Model APIs on any model in this model group.
+
+An admin can access all model groups in the cluster regardless of their access mode.
+
+
+
+## Registering a model group:
+
+User should use the `_register` endpoint to register a model group. User can register a model group with a `public`, `private`, or `restricted` access mode.
+
+### Path and HTTP method
+
+```
+POST /_plugins/_ml/model_groups/_register
+```
+
+### Request fields
+
+The following table lists the available request fields.
+
+| Field                   | Data type | Description                                                                                                                                                                                                                                               |
+|:------------------------|:----------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `name`                  | String    | The model group name. Required.                                                                                                                                                                                                                           |
+| `description`           | String    | The model group description. Optional.                                                                                                                                                                                                                    |
+| `model_access_mode`     | String    | The access mode for this model. Valid values are `public`, `private`, and `restricted`. When this parameter is set to `restricted`, user must specify either `backend_roles` or `add_all_backend_roles`, but not both. Optional. Default is `restricted`. |
+| `backend_roles`         | Array     | A list of the model owner's backend roles to add to the model. Can be specified only if the `model_access_mode` is `restricted`. Cannot be specified at the same time as `add_all_backend_roles`. Optional.                                               |
```

Comment:
Can be specified only if the `model_access_mode` is `restricted`. --> Required only if the `model_access_mode` is `restricted` ?